app/lib/frontend/backend.dart

```diff
@@ -297,6 +296,38 @@ class Backend {
       await analyzerClient.triggerAnalysis(package, latestVersion, <String>{});
     });
   }
+
+  /// Whether the current authenticated user (or [userId] if provided) is a
+  /// package admin (through direct uploaders list or publisher admin).
+  ///
+  /// Returns false if the user is not an admin.
+  Future<bool> isPackageAdmin(models.Package p, {String userId}) async {
+    userId ??= authenticatedUser?.userId;
+    if (userId == null) {
+      return false;
+    }
+    if (p.publisherId == null) {
+      return p.containsUploader(userId);
+    } else {
+      final memberKey = db.emptyKey
+          .append(Publisher, id: p.publisherId)
+          .append(PublisherMember, id: userId);
+      final list = await db.lookup<PublisherMember>([memberKey]);
+      final member = list.single;
+      return member?.role == PublisherMemberRole.admin;
```

Comment:
Can it ever not have a  `role`?

Why would we do `null` checks, if we don't allow `null`?

Reply:
`member` can be null (e.g. user is not a member).

Reply:
ah, nevermind :)